### ISSUE TITLE
Rendering manifests where canvases have identical labels

### ISSUE THREAD

**Issue by aisaac** [mejackreed/Leaflet-IIIF#52]
At Europeana we have found a problem with viewing some IIIF resources, like
http://www.europeana.eu/portal/fr/record/9200365/BibliographicResource_1000055020174.html

It seems that the problem happens when IIIF manifests have canvases with identical labels, such as this one, where all canvases have the same label (“NP”):
http://gallica.bnf.fr/iiif/ark:/12148/btv1b6000107x/manifest.json

A simple Leaflet rendering is 
http://styleguide.europeana.eu/patterns/molecules-components-iiif/molecules-components-iiif.html?manifestUrl=http://gallica.bnf.fr/iiif/ark:/12148/btv1b6000107x/manifest.json

When loading in another IIIF viewer (Universal Viewer) the problem does not appear:
http://universalviewer.io/examples/?manifest=http://gallica.bnf.fr/iiif/ark:/12148/btv1b6000107x/manifest.json

Here’s another case, where only a subset of canvases have the same label: 
http://styleguide.europeana.eu/patterns/molecules-components-iiif/molecules-components-iiif.html?manifestUrl=http://gallica.bnf.fr/iiif/ark:/12148/btv1b55005728g/manifest.json vs http://universalviewer.io/examples/?manifest=http://gallica.bnf.fr/iiif/ark:/12148/btv1b55005728g/manifest.json

In this example the ToC / order or pages is not the same as specified in the manifest. There are several canvases with “NP” label at the beginning, which may well have disappeared.

**Comment by mejackreed** [comment]
Hi @aisaac thanks for filing the ticket. Leaflet-IIIF works only with Image API uri's and doesn't aim to parse IIIF manifests. I think this problem may be with your manifest parsing code? Leaflet-IIIF doesn't look at manifest canvases and is only concerned with the info.json url.

Do you have a reference to that code?


A few examples of image api requests working:
http://mejackreed.github.io/Leaflet-IIIF/examples/?url=http://gallica.bnf.fr/iiif/ark:/12148/btv1b6000107x/f2/info.json

http://mejackreed.github.io/Leaflet-IIIF/examples/?url=http://gallica.bnf.fr/iiif/ark:/12148/btv1b6000107x/f3/info.json

Am I understanding your issue correctly?

**Comment by aisaac** [comment]
Hi @mejackreed

Thanks for the quick and useful feedback.
You are probably right about the source. I have to confess I did not know where the boundary lied between our code and Leaflet-IIIF... Neither do I know where our code is. I'm just not responsible of this area in Europeana.
I've created a ticket on our own Assembla space so that we can investigate the issue on our side:
https://app.assembla.com/spaces/europeana-npc/tickets/1661-rendering-iiif-manifests-where-canvases-have-identical-labels

We'll keep you in the loop in case anything would concern you, if it happens!

Cheers,

Antoine

**Comment by aisaac** [comment]
Hi @mejackreed

Thanks for the quick and useful feedback.
You are probably right about the source. I have to confess I did not know where the boundary lied between our code and Leaflet-IIIF... Neither do I know where our code is. I'm just not responsible of this area in Europeana.
I've created a ticket on our own Assembla space so that we can investigate the issue on our side:
https://app.assembla.com/spaces/europeana-npc/tickets/1661-rendering-iiif-manifests-where-canvases-have-identical-labels

We'll keep you in the loop in case anything would concern you, if it happens!

Cheers,

Antoine
